atest/robot/libdoc/spec_library.robot

```diff
@@ -98,6 +99,28 @@ Keyword Source Info
     Run Libdoc And Parse Output    %{TEMPDIR}/Example.libspec
     Test Everything
 
+SOURCE_DATE_EPOCH is honored in Libdoc output
+    Copy File    ${TESTDATADIR}/ExampleSpec.xml    %{TEMPDIR}/Example.libspec
+
+    TRY
+        ${current_source_date_epoch} =    Get Environment Variable    SOURCE_DATE_EPOCH
+    EXCEPT
+        ${current_source_date_epoch} =    Set Variable    ${NONE}
+    END
+
+    TRY
+        # Set the SOURCE_DATE_EPOCH to Epoch 0, i.e. 1970/01/01 at midnight GMT.
+        Set Environment Variable    SOURCE_DATE_EPOCH    0
+        Run Libdoc And Parse Output    %{TEMPDIR}/Example.libspec
+        Generated Should Be    1970-01-01T00:00:00+00:00
+    FINALLY
+        IF    $current_source_date_epoch is None
+            Remove Environment Variable    SOURCE_DATE_EPOCH
+        ELSE
+            Set Environment Variable    SOURCE_DATE_EPOCH    ${current_source_date_epoch}
+        END
+    END
```

Comment:
This is _really_ complicated. Why not just set `SOURCE_DATE_EPOCH` to a know value and validate that generation time is expected? I see no point in checking is the environment variable set earlier.

The environment variable should probably then be removed in teardown to avoid subsequent tests being affected. That obviously changes environment of the current process if `SOURCE_DATE_EPOCH` was set in the system earlier, but no acceptance test should rely on external environment variables like that. If considered important, that environment variable could be cleared automatically either by Libdoc tests or even in `atest/run.py` before starting execution.

Reply:
Clearing the value in atest/run.py should be doable, good idea. I'll try it.

Reply:
I've implemented a much simplified (albeit less clean) version in the latest push.